src/dexpo.cpp

```diff
@@ -61,39 +61,107 @@ main ()
+                auto act_width = dexpo_padding;
+                for (const auto &dexpo_pixmap : v)
+                  {
+                    xcb_copy_area (window::c_, dexpo_pixmap.id, w.id,
+                                   desktop_pixmap::gc_, 0, 0, act_width,
+                                   dexpo_padding, dexpo_pixmap.width,
+                                   dexpo_pixmap.height);
+                    act_width += dexpo_pixmap.width;
+                    act_width += dexpo_padding;
+                  };
+              }
+            else if (dexpo_height == 0)
+              {
+                auto act_height = dexpo_padding;
+                for (const auto &dexpo_pixmap : v)
+                  {
+                    xcb_copy_area (window::c_, dexpo_pixmap.id, w.id,
+                                   desktop_pixmap::gc_, 0, 0, dexpo_padding,
+                                   act_height, dexpo_pixmap.width,
+                                   dexpo_pixmap.height);
+                    act_height += dexpo_pixmap.height;
+                    act_height += dexpo_padding;
+                  };
+              }
+            w.highlight_window (0, dexpo_hlcolor);
+            break;
+          }
+        case XCB_KEY_PRESS:
+          {
+            xcb_key_press_event_t *kp
+                = reinterpret_cast<xcb_key_press_event_t *> (event);
+            ;
+            if (kp->detail == 114) // right arrow
+              {
+                highlighted += 1;
+                highlighted = highlighted % int(v.size ());
+                w.highlight_window (0,
+                                    dexpo_bgcolor); // TODO: add a function like
+                                                    // unhlight_all or smth
+                w.highlight_window (1, dexpo_bgcolor);
+                w.highlight_window (2, dexpo_bgcolor);
+                w.highlight_window (highlighted, dexpo_hlcolor);
+              }
+            if (kp->detail == 113) // left arrow
             {
-              xcb_copy_area (window::c_, dexpo_pixmap.id, w.id,
-                             desktop_pixmap::gc_, 0, 0, act_width,
-                             dexpo_padding, dexpo_pixmap.width,
-                             dexpo_pixmap.height);
-              act_width += dexpo_pixmap.width;
-              act_width += dexpo_padding;
-            };
+              if (highlighted == 0)
+                {
+                  highlighted = v.size () ;
+                }
+                  highlighted -= 1;
+            }
+            w.highlight_window (0, dexpo_bgcolor);
+            w.highlight_window (1, dexpo_bgcolor);
+            w.highlight_window (2, dexpo_bgcolor);
+            w.highlight_window (highlighted, dexpo_hlcolor);
+            if (kp->detail == 9) // escape
+              {
+                exit (0);
+              }
+            break;
+          }
+        case XCB_MOTION_NOTIFY: // pointer motion within window
```

Comment:
Add a mouse click handler (leave blank for now)

Reply:
Perhaps store all of the window's pixmap relative coordinates in a separate member struct?
Will this simplify mouse tracking? You won't have to iterate over the vectors.

Reply:
Don't preselect  desktop unless mouse is above the pixmap.
Currently desktop preselects even when the mouse pointer is near the pixmap